src/main/java/com/ylysenkova/movieland/dao/jdbc/JdbcMovieDao.java

```diff
@@ -125,5 +135,47 @@ public Movie getMovieById(int movieId) {
         return movieList;
     }
 
+    @Override
+    public void addMovie(Movie movie) {
+
+        logger.info("Inserting movie ={} is started.", movie);
+
+        KeyHolder generatedMovieId = new GeneratedKeyHolder();
+        MapSqlParameterSource sqlParameterSource = new MapSqlParameterSource();
+        sqlParameterSource.addValue("nameRussian", movie.getNameRussian());
+        sqlParameterSource.addValue("nameNative", movie.getNameNative());
+        sqlParameterSource.addValue("yearOfRelease", movie.getYearOfRelease());
+        sqlParameterSource.addValue("description", movie.getDescription());
+        sqlParameterSource.addValue("rating", movie.getRating());
+        sqlParameterSource.addValue("price", movie.getPrice());
+        sqlParameterSource.addValue("picturePath", movie.getPicturePath());
+        namedParameterJdbcTemplate.update(insertMovie, sqlParameterSource, generatedMovieId);
+        for (Genre genre : movie.getGenres()) {
+            sqlParameterSource.addValue("movieId", generatedMovieId.getKey());
+            sqlParameterSource.addValue("genreId", genre.getId());
+            namedParameterJdbcTemplate.update(insertMovieGenre, sqlParameterSource);
+        }
+        for (Country country : movie.getCountries()) {
+            sqlParameterSource.addValue("movieId", generatedMovieId.getKey());
+            sqlParameterSource.addValue("countryId", country.getId());
+            namedParameterJdbcTemplate.update(insertMovieCountry, sqlParameterSource);
+        }
+    }
```

Comment:
From my perspective would be better to persist Countries and Genres in respective Dao classes